framework/src/utils/PetscSupport.C

```diff
@@ -676,7 +682,7 @@ storePetscOptions(FEProblemBase & fe_problem, const InputParameters & params)
 #endif
 
       if (option.first == "-matptap_via" || option.first == "-matmaijptap_via" ||
-          option.first == "-matproduct_ptap_via")
+          option.first == "-matproduct_ptap_via" || option.first == "-matproduct_ptap_via")
```

Comment:
Are my eyes deceiving me? These look like the same string?

Reply:
They definitely are - should be `-mat_product_algorithm` instead. Thanks for the catch!